core/src/main/java/org/elasticsearch/action/bulk/Retry.java

```diff
@@ -102,12 +125,15 @@ public BulkResponse withSyncBackoff(Client client, BulkRequest bulkRequest) thro
         private volatile BulkRequest currentBulkRequest;
         private volatile ScheduledFuture<?> scheduledRequestFuture;
 
-        AbstractRetryHandler(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy, Client client, ActionListener<BulkResponse> listener) {
+        RetryHandler(Class<? extends Throwable> retryOnThrowable, BackoffPolicy backoffPolicy,
+                     BiConsumer<BulkRequest, ActionListener<BulkResponse>> consumer, ActionListener<BulkResponse> listener,
+                     Settings settings, BiFunction<TimeValue, Runnable, ScheduledFuture<?>> scheduleFn) {
```

Comment:
I still pass the settings here as explain in this comment https://github.com/elastic/elasticsearch/pull/23373#discussion_r109187388.